packages/ra-ui-materialui/src/field/ArrayField.spec.tsx

```diff
@@ -158,4 +174,41 @@ describe('<ArrayField />', () => {
             ).toBeTruthy();
         });
     });
+
+    it('should not select the same id in both ArrayFields when selected in one', async () => {
+        render(<TwoArrayFieldsSelection />);
+        await waitFor(() => {
+            expect(screen.queryAllByRole('checkbox').length).toBeGreaterThan(2);
+        });
+
+        const checkboxes = screen.queryAllByRole('checkbox');
+
+        expect(checkboxes.length).toBeGreaterThan(3);
+
+        // Select an item in the memberships list
+        fireEvent.click(checkboxes[1]);
+        render(<TwoArrayFieldsSelection />);
```

Comment:
You shouldn't need to rerender. If you want to come back to the initial state, use more than one `it()` and group them in a `describe()`.